private_sharing/api_views.py

```diff
@@ -270,11 +324,14 @@ class ProjectFileDirectUploadCompletionView(ProjectFormBaseView):
     form_class = DirectUploadDataFileCompletionForm
 
     def post(self, request):
-        super(ProjectFileDirectUploadCompletionView, self).post(request)
-
-        data_file = ProjectDataFile.all_objects.get(
-            pk=self.form.cleaned_data["file_id"]
-        )
+        super().post(request)
+        # If upload failed, file_id would be empty.  Let's fail gracefully.
+        file_id = self.form.cleaned_data.get("file_id", None)
+        if not file_id:
+            return Response(
```

Comment:
Seems like this hsould be raising an error the APIView handles?
e.g. https://github.com/OpenHumans/open-humans/blob/master/private_sharing/api_views.py#L160